src/Mvc/Mvc.ViewFeatures/src/ViewExecutor.cs

```diff
@@ -233,7 +233,7 @@ public class ViewExecutor
 
             OnExecuting(viewContext);
 
-            using (var writer = WriterFactory.CreateWriter(response.Body, resolvedContentTypeEncoding))
+            await using (var writer = WriterFactory.CreateWriter(response.Body, resolvedContentTypeEncoding))
```

Comment:
It should be fairly easy to write a unit test for this too. Basically we would write a test where the body of the view in https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.ViewFeatures/test/ViewExecutorTest.cs#L94-L97 throws and the test ensures that MVC does not perform a synchronous write to the HTTP response stream.